web-ts/src/hocs/withPlatform.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react'
+import Platform from '../contexts/Platform'
+import PlatformContext from '../contexts/PlatformContext'
+import wrapDisplayName from '../services/wrapDisplayName'
+
+type PropsType = {
+  platform: Platform
+}
+
+const withPlatform = <P extends PropsType>(
+  WrappedComponent: React.ComponentType<P>
+): React.ComponentType<Omit<P, keyof PropsType>> => {
+  return class extends React.Component<Omit<P, keyof PropsType>> {
+    static displayName = wrapDisplayName(WrappedComponent, 'withPlatform')
+
+    render() {
+      return (
+        <PlatformContext.Consumer>
+          {platform => <WrappedComponent {...(this.props as P)} platform={platform} />}
```

Comment:
Casting is necessary here (even though it is not entirely correct). You can check out multiple examples for that.